src/imodmodel/models.py

```diff
@@ -11,6 +11,14 @@ class ID(BaseModel):
     version_id: str
 
 
+class GeneralStorage(BaseModel):
+    """https://bio3d.colorado.edu/imod/doc/binspec.html"""
+    type: int
+    flags: int
+    index: Any
+    value: Any
```

Comment:
is it possible to type these more strictly? Maybe as a `Union[bla, bla]`?

Reply:
I had them stored as Union[int, float, Tuple[int, int], Tuple[int, int, int, int]] before but then pydantic was converting floats to int or vice versa. I guess I'd have to read the pydantic docs a bit more to understand why, but I agree this should be typed better.

Reply:
Yep automatic type coercion has some kind of order implied in the type definition - you can get around this with custom logic in a validator 🙂

Reply:
Hey,
just got the time to work on this again.
I implemented it using smart_union=True, but this has been removed in pydantic 2. I didn't notice this before because I have Napari installed in the same venv, which requires pydantic<2. 

Maybe pydantic<2 should be explicitly required.

Reply:
Agh this whole pydantic updgrade has been a bit of a nightmare - I'm fine with requiring <2 for now since your desired use case is to coexist with napari ☺️

Reply:
OK I'll add pydantic<2 as a dependency